src/solidlsp/ls_utils.py

```diff
@@ -167,16 +167,28 @@ class FileUtils:
     def read_file(logger: LanguageServerLogger, file_path: str) -> str:
         """
         Reads the file at the given path and returns the contents as a string.
+        Tries multiple encodings in order: utf-8, utf-8-sig, latin-1, cp1252
         """
         if not os.path.exists(file_path):
             logger.log(f"File read '{file_path}' failed: File does not exist.", logging.ERROR)
             raise SolidLSPException(f"File read '{file_path}' failed: File does not exist.")
-        try:
-            with open(file_path, encoding="utf-8") as inp_file:
-                return inp_file.read()
-        except Exception as exc:
-            logger.log(f"File read '{file_path}' failed to read with encoding 'utf-8': {exc}", logging.ERROR)
-            raise SolidLSPException("File read failed.") from None
+
+        encodings = ["utf-8", "utf-8-sig", "latin-1", "cp1252"]
```

Comment:
We were considering making the encoding configurable in #121 . Would that solve your use case? How did you run into this problem?